web/controllers/api/v1/message_controller.ex

```diff
@@ -6,26 +6,58 @@ defmodule Kastlex.API.V1.MessageController do
 
   plug Kastlex.Plug.EnsurePermissions
 
-  # TODO: partitioner
   def produce(conn, %{"topic" => topic, "partition" => partition} = params) do
     {partition, _} = Integer.parse(partition)
     key = Map.get(params, "key", "")
     {:ok, value, conn} = read_body(conn)
-    case :brod.produce_sync(:kastlex, topic, partition, key, value) do
-      :ok ->
-        send_resp(conn, 204, "")
+    try_produce(conn, topic, partition, key, value)
+  end
+
+  # no partition, use random partitioner
+  def produce(conn, %{"topic" => topic} = params) do
+    key = Map.get(params, "key", "")
+    {:ok, value, conn} = read_body(conn)
+    case :brod_client.get_partitions_count(:kastlex, topic) do
+      {:ok, partitions_cnt} ->
+        partitions = :lists.seq(0, partitions_cnt - 1) |> Enum.shuffle
+        try_produce(conn, topic, partitions, key, value)
       {:error, :UnknownTopicOrPartition} ->
```

Comment:
error reason -> http code mapping can be abstracted into a common function share with try_produce